packages/components/eslint.config.mjs

```diff
@@ -75,6 +75,7 @@ export default tseslint.config(
 
       "@esri/calcite-components/no-dynamic-createelement": "warn",
       "@esri/calcite-components/strict-boolean-attributes": "error",
+      "@esri/calcite-components/require-deprecation-and-removal-versions": "warn",
```

Comment:
Suggestion: renaming to `require-deprecation-details` might help future proof this rule while still being expressive and concise.